library.py

```diff
@@ -0,0 +1,207 @@
+    result = result - torch._C._dispatch_keyset_full_after(dk)
+    result = result.remove(dk)
+    return result
+
+
+def register_backend(lib, name, kernel, key):
+    if kernel is None:
+        def wrapped(*args, **kwargs):
+            raise RuntimeError(
+                "{name}: was passed {key} Tensors, but "
+                "{name}.{key.lowercase}_impl was not defined")
+    else:
+        def wrapped(*args, **kwargs):
+            before_dense = dispatch_keyset_before(torch._C.DispatchKey.Dense)
+            with torch._C._ExcludeDispatchKeyGuard(before_dense):
+                return kernel(*args, **kwargs)
+
+    lib.impl(name, wrapped, key)
+
+
+def mangle_module(module):
+    """Mangles the module name.
+
+    The scheme is replacing dots with some number of underscores
+    (specified as mangledN where N is the number of underscores).
+
+    Examples:
+    foo.bar.baz -> mangled1_foo_bar_baz
+    foo_bar.baz -> mangled2__foo_bar__baz
+    foo.__baz__ -> mangled3___foo_____baz__
+
+    Don't parse the mangled string directly; use mangle_module and demangle_module
+    """
+    sep = unique_underscoring(module)
+    prefix = f"mangled{len(sep)}"
+    splits = module.split(".")
+    return sep.join([prefix, *splits])
+
+
+def demangle_module(mangled_module):
+    pass
+
+
+def unique_underscoring(s: str):
+    i = 1
+    while True:
+        result = "_" * i
+        if result not in s:
+            return result
+        i += 1
+
+
+def check_supported_schema(opdef):
+    return
+    # TODO: figure out what to do with torchgen FunctionSchema vs C++ FunctionSchema
+
+    # We only support the following schemas, for now.
+    # - functional
+    # - auto_functionalizable.
+    # For all others, we ask the user to go use the raw torch.library API.
+    schema = opdef._schema
+    if torch._library.utils.is_functional_schema(schema):
+        return
+    if torch._higher_order_ops.auto_functionalized.auto_functionalizable_schema(schema):
+        return
+    raise NotImplementedError("")
+
+
+def register_autograd(lib, name, opdef, op):
+    class MyFunction(torch.autograd.Function):
+        @staticmethod
+        def forward(ctx, *inputs):
+            # TODO: mark-dirty things
+            with torch._C._AutoDispatchBelowAutograd():
+                output = op(*inputs)
+            if hasattr(opdef, "setup_backward"):
+                opdef.setup_backward(ctx, inputs, output)
+            return output
+
+        @staticmethod
+        def backward(ctx, *grads):
+            return opdef.backward(ctx, *grads)
+
+    lib.impl(name, MyFunction.apply, "Autograd")
+
+
+class RegistersSubclassOnDefinition(type):
+    def __new__(cls, name, bases, dct):
+        result = super().__new__(cls, name, bases, dct)
+        # This is the base class
+        if name == "Operator":
+            return result
+        opoverload = register(result)
+        result._opoverload = opoverload
+        return result
+
```

Comment:
```suggestion
        return result
        
    def __setattr__(cls, name, val):
        super().__setattr__(name, val)
        # Add any field that should not trigger a re-register
        if name not in ["_opoverload"]:
            cls._opoverload = register(result)
```

If you want any update of the class to re-trigger a register, you can do this so that `MySin.impl_cpu = foo` will work.

Reply:
:O